### PR TITLE
Updating for Crystal 1.0

### PR DESCRIPTION
**Pull request by jwoertink**
This PR is to open up the crystal version restrictions so it can be used going past 1.0. The specs fail to run locally due to `HaveFiles`, but I [opened up an issue](https://github.com/mosop/have_files/issues/7).

### CODE REVIEW

**Comment by haarts** [comment]
When that issue is resolved I believe the next one to fix is the one in [`src/lib/file_data.cr:12`](https://github.com/luckyframework/teeplate/blob/master/src/lib/file_data.cr#L12). There's a type incompatibility: I64 vs UI64. I believe the right fix is to align the instance variable `@size` with the std lib. Eg change the type from ui64 to i64.

When *that's* fixed the specs fail. I couldn't find an evident culprit.

Failure was:

```
Failures:

  1) filters templates by the rendered_filter_entries method

       Unable to create directory: '/tmp/IJtAyx/diff': File exists (File::AlreadyExistsError)
         from /usr/share/crystal/src/crystal/system/unix/dir.cr:65:7 in 'create'
         from /usr/share/crystal/src/dir.cr:234:5 in 'mkdir'
         from /usr/share/crystal/src/dir.cr:233:3 in 'mkdir'
         from /usr/share/crystal/src/file_utils.cr:128:7 in 'cp_recursive'
         from /usr/share/crystal/src/file_utils.cr:123:5 in 'cp_r'
         from lib/have_files/src/spec/expectation.cr:33:11 in 'match'
         from /usr/share/crystal/src/spec/expectations.cr:500:10 in 'should_not'
         from spec/features/filtering_file_entries_spec.cr:26:7 in '->'
         from /usr/share/crystal/src/primitives.cr:255:3 in 'internal_run'
         from /usr/share/crystal/src/spec/example.cr:33:16 in 'run'
         from /usr/share/crystal/src/spec/context.cr:18:23 in 'internal_run'
         from /usr/share/crystal/src/spec/context.cr:147:7 in 'run'
         from /usr/share/crystal/src/spec/dsl.cr:274:7 in '->'
         from /usr/share/crystal/src/primitives.cr:255:3 in 'run'
         from /usr/share/crystal/src/crystal/main.cr:45:14 in 'main'
         from /usr/share/crystal/src/crystal/main.cr:119:3 in 'main'
         from __libc_start_main
         from _start
         from ???
       
``` 

As a side note I don't see the merit of this fork. Diffing the repo's doesn't yield anything significant. Perhaps try to deprecate this fork and upstream the changes?

**Comment by jwoertink** [comment]
I'm just rolling with my own fork for now. I would love to eventually deprecate this version, but since the original isn't updated as often, using this at least allows us to move forward for now.

**Comment by matthewmcgarvey** [comment]
@jwoertink would it make sense to inline the expectation into this code base? (with attribution of course)

Seems like having one less dependency, especially as insignificant as a spec expectation helper would be a good thing

**Comment by jwoertink** [comment]
That sounds fine. I'm all for less dependencies! I'll merge my fork in to this shard.

**Comment by jwoertink** [comment]
Ok, I'm still getting this error locally

```
 1) filters templates by the rendered_filter_entries method

       Unable to create directory: '/tmp/1R6160/diff': File exists (File::AlreadyExistsError)
```

but after messing with it for the last 30 min, I'm not sure what's going on exactly. It almost seems like `FileUtils.cp_r` is returning a file instead of a directory, but I don't think it's major enough to track down for now. It seems to be specific to the specs, and not really the shard itself.

File: shard.yml [hunk 1]
```diff
@@ -4,17 +4,14 @@ version: 0.8.2
 authors:
   - mosop
 
-crystal: 0.35.0
+crystal: ">= 0.35.0, < 2.0.0"
```
Comment (matthewmcgarvey) [inline shard.yml]:
```suggestion
crystal: ">= 0.36.1, < 2.0.0"
```

**Review by matthewmcgarvey — APPROVED** [review]

**Pull request merged**